decorator.rb

```diff
@@ -0,0 +1,34 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
```

Comment:
[OPTIONAL] It would be better to print the Claas name and the method name as an error message. This will help you in the future to track the error and debug your app easily. see the example below:
```rb
class Nameable
    def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
end
```